scripts/apps/monitoring/directives/MonitoringGroup.ts

```diff
@@ -141,14 +141,24 @@ export function MonitoringGroup(
                 );
             }
 
-            function scheduleQueryFn(event, data) {
-                queryItems(event, data, {auto: (data && data.force) ? 0 : 1})
+            function scheduleQueryFn(event?, data?) {
+                return queryItems(event, data, {auto: (data && data.force) ? 0 : 1})
                     .finally(() => {
                         scope.$applyAsync();
                     });
             }
 
-            const scheduleQuery = throttle(scheduleQueryFn, 1000);
+            let queryPromise = null;
+            const scheduleQuery = debounce((event, data) => {
+                if (queryPromise == null) {
+                    queryPromise = scheduleQueryFn(event, data)
```

Comment:
I would call `queryItems` here directly and remove `scheduleQueryFn` since it doesn't do anything, except one line in `.finally`